dspace-api/src/main/java/org/dspace/app/mediafilter/MediaFilterServiceImpl.java

```diff
@@ -112,6 +117,14 @@ public void afterPropertiesSet() throws Exception {
 
     @Override
     public void applyFiltersAllItems(Context context) throws Exception {
+        boolean storeLastDate = false;
+        if (fromDate == null) {
+            storeLastDate = true;
+            String lastDate = siteService.getMetadata(siteService.findSite(context), "dspace.filtermedia.lastdate");
+            if (lastDate != null) {
```

Comment:
Additional && (isForce == false) should probably be added here to ensure that full indexing always starts if -f parameter is used. Otherwise fromDate will be set and indexing is restricted only to new items.